src/Types.jl

```diff
@@ -1020,8 +1059,33 @@ function update_registries(ctx::Context, regs::Vector{RegistrySpec} = collect_re
     !force && UPDATED_REGISTRY_THIS_SESSION[] && return
     errors = Tuple{String, String}[]
     for reg in unique(r -> r.uuid, find_installed_registries(ctx, regs))
-        if isdir(joinpath(reg.path, ".git"))
-            regpath = pathrepr(reg.path)
+        regpath = pathrepr(reg.path)
+        if isfile(joinpath(reg.path, ".tree_info.toml"))
+            printpkgstyle(ctx, :Updating, "registry at " * regpath)
+            tree_info = TOML.parsefile(joinpath(reg.path, ".tree_info.toml"))
+            old_hash = tree_info["git-tree-sha1"]
+            url = pkg_server_registry_url(reg.uuid)
+            if url !== nothing && (new_hash = pkg_server_url_hash(url)) != old_hash
+                # TODO: update faster by using a diff, if available
+                mktempdir() do tmp
+                    try
+                        download_verify_unpack(url, nothing, tmp, ignore_existence = true)
+                        tree_info_file = joinpath(tmp, ".tree_info.toml")
+                        ispath(tree_info_file) &&
+                            error("tree info file $tree_info_file already exists")
+                        open(tree_info_file, write=true) do io
+                            println(io, "git-tree-sha1 = ", repr(new_hash))
+                        end
+                        registry_file = joinpath(tmp, "Registry.toml")
+                        registry = read_registry(registry_file; cache=false)
+                        verify_registry(registry)
+                        mv(tmp, reg.path, force=true)
+                    catch err
+                        @warn "could not download $url"
```

Comment:
This is a lot of stuff inside `try catch`. Presumablyt, only the `download_verify_unpack` needs to be in it?

Reply:
Yeah, I think this will also swallow that `error()` and not print it; unsure if that is desirable.

Reply:
Yeah, these try/catch blocks are way too big; not sure what I was thinking.